### ISSUE TITLE
IDE0065 (Using directives must be placed outside of a namespace declaration) for aliases

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslyn#69251]
**Version Used**: latest

**Steps to Reproduce**:

```c#
// file 1

namespace ClassNamedAsNamespace;

public class ClassNamedAsNamespace;
```

```c#
// file 2

// using ClassNamedAsNamespace = ClassNamedAsNamespace.ClassNamedAsNamespace; this doesn't work, `ClassNamedAsNamespace obj` produces ambiguous referefce

namespace NamespaceName;

using ClassNamedAsNamespace = ClassNamedAsNamespace.ClassNamedAsNamespace; // <- IDE0065

public class MyClass
{
    public void Foo(ClassNamedAsNamespace obj)
    {
    }
}
```

**Diagnostic Id**: IDE0065 

**Expected Behavior**: No IDE0065 is produced, since I can't move this using to the top level without modifications (like inventing new neame for alias)

**Actual Behavior**: IDE0065 is reported

**Comment by CyrusNajmabadi** [comment]
> since I can't move alias using to the top level

Why can you not do this?

**Comment by hankovich** [comment]
@CyrusNajmabadi I've updated the issue with less complicated sample. I can, but I have to invent a new name for alias, I can't just use class name as is

**Comment by CyrusNajmabadi** [comment]
You'll have to suppress in that case.  if you've setup your rules taht this is an error, then it's an error.  You'll have to write references that you would normally want to go to that alias to be written in another form.  

Thanks!
